src/logic/safe/api/fetchTokenCurrenciesBalances.ts

```diff
@@ -73,15 +78,32 @@ interface BlockscoutTokenBalance {
 }
 
 async function fetchTokens(address: string) {
+  const tokenBalances = await fetchTokensWithRetriesAndAlternate(address)
+  return tokenBalances
+}
+
+type TokenAPIResult = BlockScoutApiResult<BlockScoutTokenBalance[]>
+
+async function fetchTokensWithRetriesAndAlternate(address: string) {
   const blockScoutEndpoint = `${getNetworkExplorerInfo().apiUrl}?module=account&action=tokenlist&address=${address}`
-  // retry once if it fails as gonza says that happens sometimes
-  const res = await fetch(blockScoutEndpoint).catch((e) => {
-    console.warn('Fetch error', e)
-    return fetch(blockScoutEndpoint)
-  })
-  const tokenBalances: BlockscoutTokenBalance[] = (await res.json()).result
 
-  return tokenBalances
+  for (let i = 0; i < 3; i++) {
+    const results = await Promise.allSettled<[Promise<TokenAPIResult>, Promise<TokenAPIResult>]>([
+      fetch(blockScoutEndpoint).then((res) => res.json()),
+      fetch(`https://rc1-blockscout.celo-testnet.org/api?module=account&action=tokenlist&address=${address}`).then(
+        (res) => res.json(),
+      ),
+    ])
+    const fulfilled = results.find((result) => result.status === 'fulfilled')
+    if (fulfilled) {
+      const successful = fulfilled as PromiseFulfilledResult<TokenAPIResult>
+      return successful.value.result
+    }
+  }
+
+  console.error('fetching tokens failed')
+
+  return []
```

Comment:
Previously this returned whatever tokenBalances was even if it wasn't "fulfilled", but now we're returning an empty array if these fail. Do we need to account for the error return value?

Reply:
if successful it returns a array of objects. so when not successfull an empty array means the type is the same.